lib/libzfs/libzfs_iter.c

```diff
@@ -141,7 +141,7 @@ zfs_iter_filesystems(zfs_handle_t *zhp, zfs_iter_f func, void *data)
  */
 int
 zfs_iter_snapshots(zfs_handle_t *zhp, boolean_t simple, zfs_iter_f func,
-    void *data)
+    void *data, nvlist_t *range_nvl)
```

Comment:
We don't care about changing this interface and breaking 3rd party libzfs consumers, right?  They should be using libzfs_core, the CLI, or the kernel interface directly if they want a stable interface?

Reply:
hmm good point; I could create zfs_iter_snapshots_impl() and use that so that we leave the signature for zfs_iter_snapshots() as is but it sounds like that's not necessary